src/main/java/com/o19s/es/ltr/query/FeatureVectorWeight.java

```diff
@@ -49,10 +55,28 @@ public Explanation explain(LeafReaderContext context, int doc) throws IOExceptio
     public abstract Explanation explain(LeafReaderContext context, LtrRanker.FeatureVector vector, int doc) throws IOException;
 
     /**
-     *
-     * @param context surrent segment
+     * @param context        surrent segment
      * @param vectorSupplier supplier of the feature vector (always call get(), it must not be cached)
      * @return The scorer
      */
     public abstract Scorer scorer(LeafReaderContext context, Supplier<LtrRanker.FeatureVector> vectorSupplier) throws IOException;
+
+    public ScorerSupplier scorerSupplier(LeafReaderContext context) throws IOException {
```

Comment:
not entirely sure that this is needed, since the vectorSupplier is now available as a class member.